MaterialLib/SolidModels/PhaseFieldExtension.h

```diff
@@ -0,0 +1,94 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2017, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include "ProcessLib/Deformation/BMatrixPolicy.h"
+#include "ProcessLib/Parameter/SpatialPosition.h"
+#include "MechanicsBase.h"
+
+namespace MaterialLib
+{
+namespace Solids
+{
+
+template <int DisplacementDim>
+struct PhaseFieldExtension : public MechanicsBase<DisplacementDim>
+{
+    using KelvinVector = ProcessLib::KelvinVectorType<DisplacementDim>;
+    using KelvinMatrix = ProcessLib::KelvinMatrixType<DisplacementDim>;
+    virtual bool calculateDegradedStress(double const t,
+                                 ProcessLib::SpatialPosition const& x,
+                                 KelvinVector const& eps,
+                                 double& strain_energy_tensile,
+                                 KelvinVector& sigma_tensile,
+                                 KelvinVector& sigma_compressive,
+                                 KelvinMatrix& C_tensile,
+                                 KelvinMatrix& C_compressive,
+                                 KelvinVector& sigma_real,
+                                 double const degradation) const = 0;
+
+    /// Dynamic size Kelvin vector and matrix wrapper for the polymorphic
+    /// constitutive relation compute function.
+    // Decompose the stiffness into tensile and compressive part.
+    // Judging by the physical observations, compression perpendicular
+    // to a crack does not cause crack propagation. Thus,
+    // the phase-field parameter is only involved into the tensile part
+    // to degrade the elastic strain energy.
```

Comment:
Ah, I see the description is at wrong place. Please move it up to the previously undocumented function. This function was already documented.